File: src/features/typing/data/ts/rotate_array.ts
function rotate(nums: number[], k: number): void {
    const n = nums.length;
    k %= n;
    const reverse = (lo: number, hi: number): void => {
        for (let i = lo, j = hi; i < j; i++, j--) {
            const tmp = nums[i];
            nums[i] = nums[j];
            nums[j] = tmp;
        }
    };
    reverse(0, n - 1);
    reverse(0, k - 1);
    reverse(k, n - 1);
}

File: src/features/typing/data/go/rotate_array.go
package main

func rotate(nums []int, k int) {
	k %= len(nums)
	reverse := func(a []int) {
		for i, j := 0, len(a)-1; i < j; i, j = i+1, j-1 {
			a[i], a[j] = a[j], a[i]
		}
	}
	reverse(nums)
	reverse(nums[:k])
	reverse(nums[k:])
}
